DOCS.md

```diff
@@ -3,7 +3,14 @@
 Snippets are formatted as follows:
 
 ```typescript
-{trigger: string | RegExp, replacement: string, options: string, priority?: number, description?: string, flags?: string}
+{
+  trigger: string | RegExp,
+  replacement: string | ((str: string) => string) | ((match: RegExpExecArray) => string) | ((selection: string) => (string | false)),
+  options: string,
+  priority?: number,
+  description?: string,
+  flags?: string,
+}
```

Comment:
Showing the full type definition for the `replacement` here will probably be overwhelming for users. My suggestion would be we leave this as `replacement: string`, and then create a new section lower in the docs titled "Function snippets", where we can show the full type definition, as well as include example snippets etc.
  
 Under the "Function snippets" heading I'd have something like
```md
Replacements can also be functions. **Function snippets** take the form {insert full type definition here}.
```
I'd be happy to help write the docs, by the way.

Reply:
added a section for `Function snippets` after snippet variables in `DOCS.md` and reverted type of `replacement` to `string` in the readme / snippet overview.